app/src/main/kotlin/compiler/lexer/lexer_grammar/RegexParser.kt

```diff
@@ -3,10 +3,25 @@ package compiler.lexer.lexer_grammar
 import compiler.lexer.regex.Regex
 import compiler.lexer.regex.RegexFactory
 
-class RegexParser {
+class RegexParser : UniversalRegexParser<Regex>() {
```

Comment:
It would be problematic at the moment since RegexParser is used in the Tokens class, which will be implemented by someone else, and changing it could create merge conflicts. Maybe I will do it after all current PRs are merged?

Reply:
It’s a matter of removing parenthesis  in a few places. I believe that we can handle such conflict.